BrainPortal/app/models/tool_config.rb

```diff
@@ -326,22 +326,45 @@ def cbrain_task_class
 
   # Returns an array of full paths to the Singularity overlay files that
   # need to be mounted, as configured by the admin. Some of them might
-  # be patterns and will need to be resolved at run time.
+  # be patterns and will need to be resolved at run time. The dsl is
+  #      # A file located on the cluster
+  #        file:/path/to/file.squashfs
+  #      # All overlays configured for DP 1234
+  #         dp:1234
+  #      # My special overlay in CBRAIN
+  #         userfile:1234
   def singularity_overlays_full_paths
     specs = self.singularity_overlays_specs.presence
     return [] if specs.blank?
 
-    specs = specs.split(/[\s,]+/).map(&:presence).compact
+    # break onto commands based on comas, whitespace or comments
+    specs = specs.split(/([\s,]|#.*$)+/).map(&:presence).compact
     specs.map do |spec|
-      # Full path or pattern
-      next spec if spec =~ /^\// # just use that
+      # Full path or pattern (legacy)
+      next spec if spec =~ /^\// # just use that - FIXME remove after successful migration
+      cb_error "Spec #{spec}  starts with semicolon" if spec.start_with?(':')
```

Comment:
Why all the low level string logic? Just match! Something like

```ruby
cb_error "bad spec message here" unless spec =~ /\A(file|dp|userfile):(\S+)\s*\z/
scheme, id_or_name = Regexp.last_match[1,2]
```

Reply:
better diagnosis. 

but if not needed can use that as well.